src/Android/Xamarin.Android/ApiKeyPrompt.cs

```diff
@@ -0,0 +1,101 @@
+﻿// Copyright 2021 Esri.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at: http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific
+// language governing permissions and limitations under the License.
+
+using Android.App;
+using Android.OS;
+using Android.Text.Method;
+using Android.Widget;
+using ArcGISRuntime.Samples.Shared.Managers;
+using System;
+using System.Threading.Tasks;
+
+namespace ArcGISRuntime
+{
+    [Activity(Label = "ApiKeyPrompt")]
+    public class ApiKeyPrompt : Activity
+    {
+        private Button _setKeyButton;
+        private Button _deleteKeyButton;
+        private Button _storeKeyButton;
+
+        private TextView _currentKeyText;
+        private TextView _statusText;
+
+        private EditText _keyEntry;
+
+        protected override void OnCreate(Bundle savedInstanceState)
+        {
+            base.OnCreate(savedInstanceState);
+
+            // Setup the UI.
+            SetContentView(Resource.Layout.ApiKeyPrompt);
+
+            var linkTextView = FindViewById<TextView>(Resource.Id.ApiKeyInstructions);
+            linkTextView.MovementMethod = LinkMovementMethod.Instance;
+
+            _setKeyButton = FindViewById<Button>(Resource.Id.setKeyButton);
+            _deleteKeyButton = FindViewById<Button>(Resource.Id.deleteKeyButton);
+            _storeKeyButton = FindViewById<Button>(Resource.Id.storeKeyButton);
+
+            _setKeyButton.Click += SetKey;
```

Comment:
I'm reviewing this again, and I don't really understand why the viewer doesn't immediately persist the API key once set. Could we get rid of the separate remember button and save the updated key immediately once set/deleted?